requirements/tests.txt

```diff
@@ -10,4 +10,4 @@ pytest-randomly>=3.12.0
 git+https://github.com/elastic/perf8#egg=perf8
 freezegun>=0.3.4
 pytest-fail-slow>=0.3.0
-pyright>=1.1.295
+pyright>=1.1.312
```

Comment:
I actually had to do it in this, because my usage of `zip` didn't work with the old version.  https://github.com/microsoft/pyright/issues/5172#issuecomment-1579677316